src/components/PopupWithForm.js

```diff
@@ -7,6 +7,7 @@ class PopupWithForm extends Popup {
       this._handleFormSubmit = handleFormSubmit;
       this._submitButton = this._popupElement.querySelector('.modal__button')
       this._formInputs = this._popupForm.querySelectorAll(".modal__input");
+      this._submitButtonText = this._submitButton.textContent;
     }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`